File: branches/Medusa/Source/Fantasy.Base/Tracking/TrackingConfiguration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;
using System.Threading;

namespace Fantasy.Tracking
{
    public static  class TrackingConfiguration
    {

        private static List<ServiceHost> _hosts = new List<ServiceHost>();

        public static void StartTrackingService()
        {

            Thread thread = ThreadFactory.CreateThread(()=>{
            _hosts.AddRange(new ServiceHost[] 
            {
                new ServiceHost(typeof(TrackProviderService)), 
                new ServiceHost(typeof(TrackManagerService)), 
                new ServiceHost(typeof(TrackListenerService))
            });

            foreach (ServiceHost host in _hosts)
            {
                host.Open();
            }}).WithStart();
           
          
            thread.Join();
            
        }

        public static void CloseTrackingService()
        {
            foreach (ServiceHost host in _hosts)
            {
                host.Close();
            }
        }
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/Tracking/TrackingConfiguration.java
package Fantasy.Tracking;

public final class TrackingConfiguration
{

	private static java.util.ArrayList<ServiceHost> _hosts = new java.util.ArrayList<ServiceHost>();

	public static void StartTrackingService()
	{

//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		Thread thread = ThreadFactory.CreateThread(()=>{
			_hosts.addAll(new ServiceHost[] { new ServiceHost(TrackProviderService.class), new ServiceHost(TrackManagerService.class), new ServiceHost(TrackListenerService.class) });
			for (ServiceHost host : _hosts)
			{
				host.Open();
			}
	}
   ).WithStart();


		thread.Join();

	}

	public static void CloseTrackingService()
	{
		for (ServiceHost host : _hosts)
		{
			host.Close();
		}
	}
}